### PR TITLE
updates the clap config

### PR DESCRIPTION
**Pull request by kbknapp**
This commit does a few things to improve the current config:

* It makes a subcommand required
* It disables the help subcommand (because `cargo-thanks help thanks` isn't very useful to a cargo plugin
* It manually sets the binary name to `cargo` so that `--help` usage suggestions correctly say `cargo thanks` and not `cargo-thanks thanks`
* It removes the leading tabs from the about description
* It adds a `--token <token>` option which automatically gets set by the `GITHUB_TOKEN` env var
* It adds a validation function to the `--token` argument to ensure a token was provided and is not blank (such as `GITHUB_TOKEN=`
* Since clap verifies that a value was provided either by the `GITHUB_TOKEN` or `--token` argument it's safe to unwrap the value
* The CLI overrides the `GITHUB_TOKEN` var, so if someone has `GITHUB_TOKEN` set, but then also provides `--token <token>` the CLI takes precedence

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -45,28 +44,40 @@ error_chain! {
...
+            SubCommand::with_name("thanks")
+            .version(env!("CARGO_PKG_VERSION"))
+            .author(env!("CARGO_PKG_AUTHORS"))
+            .about(
+                "Thanks rust lang dependencies on github.com \n\
+                this program assumes a github token stored in a GITHUB_TOKEN env variable"
             )
+            .arg(Arg::from_usage("-t, --token [TOKEN] 'The Github OAuth token to use'")
+                .required(true)
+                .env("GITHUB_TOKEN")
```
Comment (softprops) [inline src/main.rs:71]: very cool! I did not know about these clap features

**Comment by softprops** [comment]
published a new version with these changes. very nice. 

one caveat of the convenience is security. I noticed clap now prints out my github token when I run `cargo help thanks` that's okay. I'll just have to be careful in demos!

```
 cargo help thanks
cargo-thanks 0.1.2
softprops <d.tangren@gmail.com>
Thanks rust lang dependencies on github.com
this program assumes a github token stored in a GITHUB_TOKEN env variable

USAGE:
    cargo thanks --token <TOKEN>

FLAGS:
    -h, --help       Prints help information
    -V, --version    Prints version information

OPTIONS:
    -t, --token <TOKEN>    The Github OAuth token to use [env: GITHUB_TOKEN=xxx]
```

**Comment by kbknapp** [comment]
I'm about to get on a long plane ride so I'll implement the ability to not
display the contents of the ENV far (if any). ;)

**Comment by softprops** [comment]
No worries and no rush. I wasn't really expecting a response. Now I have this mental image of you on planes, sky diving ( your avatar ) while writing rust at the same time!

**Comment by kbknapp** [comment]
:laughing: maybe one day I will write some altimeter software so I can at least be *using* Rust while skydiving.

**Pull request merged**